docs/guides/intro-to-graphql.md

```diff
@@ -1,9 +1,407 @@
+
+The options available in a where clause depend on the field types.
+
+```gql
+query {
+  allUsers (where: { id_starts_with_i: 'A'} ) {
+    id
+  }
+}
+```
+
+##### Relationship `where` filters
+
+- `{relatedList}_every`: whereInput
+- `{relatedList}_some`: whereInput
+- `{relatedList}_none`: whereInput
+- `{relatedList}_is_null`: Boolean
+
+##### String `where` filters
+
+- `{Field}:` String
+- `{Field}_not`: String
+- `{Field}_contains`: String
+- `{Field}_not_contains`: String
+- `{Field}_starts_with`: String
+- `{Field}_not_starts_with`: String
+- `{Field}_ends_with`: String
+- `{Field}_not_ends_with`: String
+- `{Field}_i`: String
+- `{Field}_not_i`: String
+- `{Field}_contains_i`: String
+- `{Field}_not_contains_i`: String
+- `{Field}_starts_with_i`: String
+- `{Field}_not_starts_with_i`: String
+- `{Field}_ends_with_i`: String
+- `{Field}_not_ends_with_i`: String
+- `{Field}_in`: [String]
+- `{Field}_not_in`: [String]
+
+##### ID `where` filters
+
+- `{Field}`: ID
+- `{Field}_not`: ID
+- `{Field}_in`: [ID!]
+- `{Field}_not_in`: [ID!]
+
+##### Integer `where` filters
+
+- `{Field}: Int`
+- `{Field}_not`: Int
+- `{Field}_lt`: Int
+- `{Field}_lte`: Int
+- `{Field}_gt`: Int
+- `{Field}_gte`: Int
+- `{Field}_in`: [Int]
+- `{Field}_not_in`: [Int]
+
+#### Operators
+
+You can combine multiple where clauses with `AND` or `OR` operators.
+
+- `AND`: [whereInput]
+- `OR`: [whereInput]
+
+#### `search`
+
+Will search the list to limit results.
+
+```gql
+query {
+  allUsers(search: "Mike") {
+    id
+  }
+}
+```
+
+#### `orderBy`
+
+Order results. The orderBy string should match a field name in the List.
+
+```gql
+query {
+  allUsers(orderBy: "name") {
+    id
+  }
+}
+```
+
+#### `first`
+
+Limits the number of items returned from the query. Limits will be applied after `skip`, `orderBy`, `where` and `search` values are applied.
+
+If less results are available, the number of available results will be returned.
+
+```gql
+query {
+  allUsers(first: 10) {
+    id
+  }
+}
+```
+
+#### `skip`
+
+```gql
+query {
+  allUsers(skip: 10) {
+    id
+  }
+}
+```
+
+Skip the number of results specified. Is applied before `first` parameter, but after `orderBy`, `where` and `search` values.
+
+If the value of `skip` is greater than the number of available results, zero results will be returned.
+
+## Custom queries and Mutations
+
+You can add to the keystones generated schema with custom types, queries, and mutations using the `keystone.extendGraphQLSchema` method.
+
+### Usage
+
+```javascript
+keystone.extendGraphQLSchema({
+  types: ['type FooBar { foo: Int, bar: Float }'],
+  queries: [
+    {
+      schema: 'double(x: Int): Int',
+      resolver: (_, { x }) => 2 * x,
+    },
+  ],
+  mutations: [
+    {
+      schema: 'double(x: Int): Int',
+      resolver: (_, { x }) => 2 * x,
+    },
+  ],
+});
+```
+
+### Combining query arguments for pagination
```

Comment:
Should this section be swapped with `Custom queries and Mutations` so it comes immediately after the descriptions of `skip` and `first`, giving it a better top-to-bottom flow in the document?

Reply:
Sure!